src/Merge/MultiConstraint.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/**
+ * This file is part of the Composer Merge plugin.
+ *
+ * Copyright (C) 2015 Bryan Davis, Wikimedia Foundation, and contributors
+ *
+ * This software may be modified and distributed under the terms of the MIT
+ * license. See the LICENSE file for details.
+ */
+
+namespace Wikimedia\Composer\Merge;
+
+use Composer\Semver\Constraint\EmptyConstraint;
+use Composer\Semver\Constraint\MultiConstraint as SemverMultiConstraint;
+
+/**
+ * Adapted from Composer's v2 MultiConstraint::create for Composer v1
+ * @link https://github.com/composer/semver/blob/3.2.4/src/Constraint/MultiConstraint.php
+ * @author Chauncey McAskill <chauncey@mcaskill.ca>
+ */
+class MultiConstraint extends SemverMultiConstraint
+{
+    /**
+     * Tries to optimize the constraints as much as possible, meaning
+     * reducing/collapsing congruent constraints etc.
+     * Does not necessarily return a MultiConstraint instance if
+     * things can be reduced to a simple constraint
+     *
+     * @param ConstraintInterface[] $constraints A set of constraints
+     * @param bool                  $conjunctive Whether the constraints should be treated as conjunctive or disjunctive
+     *
+     * @return ConstraintInterface
+     */
+    public static function create(array $constraints, $conjunctive = true)
+    {
+        if (0 === \count($constraints)) {
```

Comment:
Why is there a mix of yoda conditionals and non yoda conditionals (like just below)?

Reply:
I copied this class from composer/semver and have tried to keep changes to a minimum to make importing future updates easier.